index.js

```diff
@@ -19,49 +20,62 @@ const SUBMISSION = [
 ]
 
 app.post('/signup', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
-
-
-  //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
-
-
-  // return back 200 status code to the client
-  res.send('Hello World!')
+ 
+    const { email, password } = req.body;
+  const existingUser = USERS.find(user => user.email === email);
+  if (existingUser) {
+    return res.status(400).send('User already exists');
+  }
+  USERS.push({ email, password });
+  res.sendStatus(200);
 })
 
 app.post('/login', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
-
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
+     const { email, password } = req.body;
+    
+    const user = USERS.find(user => user.email === email);
+    if (!user) {
+      return res.status(401).send('Invalid email or password');
+    }
+   
+    if (user.password !== password) {
+      return res.status(401).send('Invalid email or password');
+    }
+   
+    const token = 'randomToken'; 
+    res.status(200).send({ token });
 
-
-  // If the password is the same, return back 200 status code to the client
-  // Also send back a token (any random string will do for now)
-  // If the password is not the same, return back 401 status code to the client
-
-
-  res.send('Hello World from route 2!')
 })
 
 app.get('/questions', function(req, res) {
-
-  //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
+  res.send(QUESTIONS)
 })
 
 app.get("/submissions", function(req, res) {
-   // return the users submissions for this problem
-  res.send("Hello World from route 4!")
+  res.send(SUBMISSION)
 });
 
 
 app.post("/submissions", function(req, res) {
-   // let the user submit a problem, randomly accept or reject the solution
-   // Store the submission in the SUBMISSION array above
-  res.send("Hello World from route 4!")
+   const { email, questionTitle, solution } = req.body;
+    const question = QUESTIONS.find(question => question.title === questionTitle);
+    if (!question) {
+      return res.status(404).send('Question not found');
+    }
+  
+    const testCase = question.testCases[0]; 
+    const expectedOutput = testCase.output;
+    const actualOutput = solution; 
+    const isCorrect = expectedOutput === actualOutput;
+  
+    SUBMISSION.push({ email, questionTitle, solution, isCorrect });
+  
+    
+    if (isCorrect) {
+      res.send('Correct solution');
+    } else {
+      res.send('Incorrect solution');
+    }
 });
 
 // leaving as hard todos
```

Comment:
@yashhhh04 in case you missed this, you can try adding a route that let's only admin to post a question